src/transformers/utils/quantization_config.py

```diff
@@ -418,3 +428,53 @@ def post_init(self):
                     f"""dataset needs to be either a list of string or a value in
                     ['wikitext2','c4','c4-new','ptb','ptb-new'], but we found {self.dataset}"""
                 )
+        if self.bits == 4:
+            if self.disable_exllama is not None:
+                logger.warning(
+                    "Using `disable_exllama` is deprecated and will be removed in version 4.37. Use `use_exllama` instead and specify the version with `exllama_version`."
+                    "The value of `use_exllama` will be overwritten by `disable_exllama` passed in `GPTQConfig` or stored in your config file."
+                )
+                self.use_exllama = not self.disable_exllama
+
+            if self.use_exllama:
+                if self.exllama_version == ExllamaVersion.ONE:
+                    logger.warning(
+                        "You have activated exllama backend. Note that you can get better inference "
+                        "speed using exllamav2 kernel by setting `exllama_version=2`."
+                    )
+                if self.exllama_version == ExllamaVersion.TWO:
+                    optimum_version = version.parse(importlib.metadata.version("optimum"))
+                    autogptq_version = version.parse(importlib.metadata.version("auto_gptq"))
+                    if optimum_version <= version.parse("1.13.2") or autogptq_version <= version.parse("0.4.2"):
+                        raise ValueError(
+                            f"You need optimum > 1.13.2 and auto-gptq > 0.4.2 . Make sure to have that version installed - detected version : optimum {optimum_version} and autogptq {autogptq_version}"
+                        )
+
+    def to_dict_optimum(self):
```

Comment:
If we're deprecating the `disable_exllama` argument, then we need to implement `to_dict` for this class, removing it so it's not save out

```
def to_dict(self):
    config_dict = super().to_dict()
    config_dict.pop("disable_exllama", None)
    return config_dict
```

Reply:
Indeed ! thx